the-collab-lab/tcl-71-smart-shopping-list
# Bm vy mark as purchased

## Description

Allow users to mark items as purchased, updating `dateLastPurchased` and `totalPurchases` on the Firestore and checking off the item for 24h.

## Related Issue

closes #9

## Acceptance Criteria


## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓    | :sparkles: New feature     |

## Updates

### Before

![image](https://github.com/the-collab-lab/tcl-71-smart-shopping-list/assets/86715948/1d8ff8e5-0649-4faf-9b73-0160b46c9f71)

### After

![image](https://github.com/the-collab-lab/tcl-71-smart-shopping-list/assets/86715948/3d84e60c-892b-4c83-bab8-20c49eb41f03)

## Testing Steps / QA Criteria

* git pull origin bm-vy-mark-as-purchased
* Navigate to /manage-list and add some items to any list using the form if you haven't already.
* Navigate to /list and mark some items as purchased.
* Open the Firebase console, check that `dateLastPurchased` and `totalPurchases` are updated, and change `dateLastPurchased` of one or more items marked as purchased.
* Navigate to /list and observe the changes in the UI.

## TODO

A functionality that could be added further down the line is the possibility of reverting the purchased state of an item in case of user error.
